chunks 1,8: Do not set date to UNDATED (to be pushed upstream).
chunk 2: Tweak page headers (local change).
chunk 3: Put architecture name in parentheses (to be pushed upstream).
chunk 4: Default value for the .Os macro (local change).
chunks 5-7: No parentheses if the section is empty (to be pushed upstream).
chunk 9: Do not use an ugly double dash in the NAME line (local change).

Index: tmac/mdoc/doc-common
--- tmac/mdoc/doc-common.orig
+++ tmac/mdoc/doc-common
@@ -278,7 +278,7 @@
 .\" NS doc-date-string global string
 .\" NS   the manual page date as set by 'Dd'
 .
-.ds doc-date-string UNDATED\"
+.ds doc-date-string \" empty
 .
 .
 .\" NS Dd user macro (not parsed, not callable)
@@ -345,7 +345,7 @@
 .ds doc-volume-ds-1 General Commands Manual
 .ds doc-volume-ds-2 System Calls Manual
 .ds doc-volume-ds-3 Library Functions Manual
-.ds doc-volume-ds-4 Kernel Interfaces Manual
+.ds doc-volume-ds-4 Device Drivers Manual
 .ds doc-volume-ds-5 File Formats Manual
 .ds doc-volume-ds-6 Games Manual
 .ds doc-volume-ds-7 Miscellaneous Information Manual
@@ -464,12 +464,11 @@
 .    ds doc-section \$2
 .    ie \B'\$2' \{\
 .      if ((\$2 >= 1) & (\$2 <= 9)) \{\
-.        ds doc-volume \" empty (not "LOCAL")
+.        ds doc-volume "\*[doc-volume-ds-\$2]
 .        if \A'\$3' \{\
 .          if d doc-volume-as-\$3 \
-.            as doc-volume "\*[doc-volume-as-\$3]
-.          \}
-.        as doc-volume " \*[doc-volume-ds-\$2]
+.            as doc-volume " (\*[doc-volume-as-\$3])
+.        \}
 .      \}
 .    \}
 .    el \{\
@@ -499,7 +498,7 @@
 .\" NS
 .\" NS override this in 'mdoc.local', if necessary
 .
-.ds doc-default-operating-system GNU\"
+.ds doc-default-operating-system OpenBSD\"
 .
 .
 .\" NS doc-operating-system global string
@@ -1009,7 +1008,8 @@
 .eo
 .de1 doc-header
 .  ds doc-xref \f[\*[doc-page-identifier-font]]\*[doc-page-identifier]\f[]\"
-.  as doc-xref \f[\*[doc-page-section-font]](\*[doc-section])\f[]\"
+.  if !"\*[doc-section]"" \
+.    as doc-xref \f[\*[doc-page-section-font]](\*[doc-section])\f[]\"
 .  ds doc-abbv \*[doc-page-identifier]\"
 .  ev doc-env-dh
 .  doc-setup-page-layout
@@ -1020,7 +1020,8 @@
 .  if (\n[doc-reg-dh] + \n[doc-reg-dh1] + \n[doc-reg-dh] >= \n[.lt]) \{\
 .    while (\n[doc-reg-dh] + \n[doc-reg-dh1] + \n[doc-reg-dh] >= \n[.lt]) \{\
 .      ds doc-xref \f[\*[doc-page-identifier-font]]\*[doc-abbv]\f[]\"
-.      as doc-xref \f[\*[doc-page-section-font]](\*[doc-section])\f[]\"
+.      if !"\*[doc-section]"" \
+.        as doc-xref \f[\*[doc-page-section-font]](\*[doc-section])\f[]\"
 .      length doc-abbv-len-prev \*[doc-abbv]
 .      substring doc-abbv 0 -2
 .      length doc-abbv-len \*[doc-abbv]
@@ -1065,7 +1066,8 @@
 .eo
 .de1 doc-footer
 .  ds doc-xref \f[\*[doc-page-identifier-font]]\*[doc-page-identifier]\f[]\"
-.  as doc-xref \f[\*[doc-page-section-font]](\*[doc-section])\f[]\"
+.  if !"\*[doc-section]"" \
+.    as doc-xref \f[\*[doc-page-section-font]](\*[doc-section])\f[]\"
 .  ds doc-page-id \n[%]
 .  if r X \{\
 .    if (\n[%] > \n[X]) \{\
@@ -1122,7 +1124,7 @@
 .eo
 .de doc-reset-titles
 .  \" Reset strings to reduce info leaks from one man page to the next.
-.  ds doc-date-string UNDATED\"
+.  ds doc-date-string \" empty
 .  ds doc-page-identifier UNTITLED\"
 .  ds doc-volume LOCAL\"
 .  ds doc-section \" empty
@@ -1336,7 +1338,7 @@
 .de Nd
 .  doc-reset-adjustment-mode
 .  doc-reset-hyphenation-mode
-.  nop \[em] \$*
+.  nop \[en] \$*
 ..
 .ec
 .
